python/cudf/cudf/tests/test_column.py

```diff
@@ -554,6 +554,13 @@ def test_astype_with_aliases(alias, expect_dtype, data):
     assert_eq(pd_data.astype(expect_dtype), gd_data.astype(alias))
 
 
+def test_as_column_None_as_object_type():
+    result = as_column(None, length=1)
+    assert isinstance(result, cudf.core.column.string.StringColumn)
```

Comment:
In relation to https://github.com/rapidsai/cudf/pull/14614 and to align with the pandas 2.0 compatibility effort, yes this was to ensure closeness to "object" column for an uptyped scalar. Since this decision can be resolved at a later point I'll remove it now from this refactor